internal/testutil/incrementaltestutil/readablebuildinfo.go

```diff
@@ -174,6 +182,27 @@ type readableBuildInfoEmitSignature struct {
 	Original            *incremental.BuildInfoEmitSignature `json:"original,omitzero"`
 }
 
+type readableBuildInfoResolvedRoot struct {
+	Resolved string
+	Root     string
+}
+
+func (b *readableBuildInfoResolvedRoot) MarshalJSON() ([]byte, error) {
+	return json.Marshal([2]string{b.Resolved, b.Root})
+}
+
+func (b *readableBuildInfoResolvedRoot) UnmarshalJSON(data []byte) error {
+	var resolvedAndRoot *[2]string
+	if err := json.Unmarshal(data, &resolvedAndRoot); err == nil {
```

Comment:
Why is resolvedAndRoot a pointer here? It seems like it should just be `[2]string`, since you're basically just passing in `**[2]string` and then assuming it decodes. It seems like maybe it's intending to handle null, but then it unconditionally uses resolvedAndRoot anyway.